### PR TITLE
feat: Variable scopes

### PR DESCRIPTION
**Pull request by doosuu**
Adds scope identifiers to variable definitions in toolchain components.

### CODE REVIEW

File: src/modules/project-config.ts [hunk 1]
```diff
@@ -191,6 +184,21 @@ export class ProjectConfig {
         return componentContexts;
     }
 
+    /**
+     * Find a single component by its ID.
+     * @param componentId   The component ID to find.
+     * @returns The context the component is used in.
+     */
+    findComponentByName(componentId: string): ComponentContext {
+        let result = this.getComponents().find((c) => c.manifest.id === componentId);
```
Comment (dennismeister93) [inline src/modules/project-config.ts:193]: type assertion (c)

File: test/commands/create/create.test.ts [hunk 2]
```diff
@@ -93,7 +93,6 @@ describe('create', () => {
             expect(velocitasConfig.getPackages()[0].version).to.be.equal(TEST_MAIN_PACKAGE_VERSION);
             expect(velocitasConfig.getPackages()[1].repo).to.be.equal(TEST_PACKAGE_URI);
             expect(velocitasConfig.getPackages()[1].version).to.be.equal(TEST_PACKAGE_VERSION);
-            expect(velocitasConfig.getVariableMappings().get('language')).to.be.equal('test');
```
Comment (dennismeister93) [inline test/commands/create/create.test.ts:96]: if we can get rid of this than you also may adapt the velocitasConfigMock inside mockConfig.ts

**Review by dennismeister93 — CHANGES_REQUESTED** [review]

File: test/commands/create/create.test.ts (hunk 2 above)
Comment (doosuu) [reply test/commands/create/create.test.ts:96]: Done.

File: src/modules/project-config.ts (hunk 1 above)
Comment (doosuu) [reply src/modules/project-config.ts:193]: Done

File: src/modules/project-config.ts [hunk 3]
```diff
@@ -175,14 +173,9 @@ export class ProjectConfig {
             const packageManifest = packageConfig.readPackageManifest();
 
             for (const componentManifest of packageManifest.components) {
-                if (usedComponents.length === 0 || usedComponents.find((compCfg: ComponentConfig) => compCfg.id === componentManifest.id)) {
+                if (usedComponents.length === 0 || usedComponents.find((c) => c.id === componentManifest.id)) {
```
[the remark this answers is not in the document]
Comment (doosuu) [reply src/modules/project-config.ts:176]: done

File: src/modules/variables.ts [hunk 4]
```diff
@@ -45,38 +78,69 @@ export class VariableCollection {
...
+                        throw Error(
+                            `Constant variable '${variableDef.name}' provided by component '${component.manifest.id}' has been set already!`,
+                        );
+                    }
+
+                    if (!map.has(variableDef.name) && variableDef.default !== undefined) {
+                        map.set(variableDef.name, variableDef.default);
+                    }
+
+                    usedVariableDefinitions.push(variableDef);
```
[the remark this answers is not in the document]
Comment (doosuu) [reply src/modules/variables.ts]: done

**Review by dennismeister93 — APPROVED** [review]
lgtm

**Pull request merged**